ports/z88dk.c

```diff
@@ -0,0 +1,198 @@
+
+//  zcc +zx81 -clib=wrx -subtype=_wrx -pragma-define:hrgpage=40000 -create-app -DGFXSCALEX=4/5 -DGFXSCALEY=4/5 -lm -O3 -DPLATFORM_MAZE=z88dk c64maze.c ports/z88dk.c
+
+// zcc +zx -lndos -create-app -DGFXSCALEX=4/5 -DGFXSCALEY=4/5 -lm -DPLATFORM_MAZE=z88dk c64maze.c ports/z88dk.c
+
+
+#include "stdio.h"
+#include "graphics.h"
+#include "games.h"
+#include "time.h"
+#include "conio.h"
+#include <bgi.h>
+
+#define	clear_area(a,b,c,d)	stencil_init(bgi_stencil);stencil_add_side((a),(b),(a),(d),bgi_stencil);stencil_add_side((c),(b),(c),(d),bgi_stencil);stencil_render(bgi_stencil,0)
+
+//#include "../vic_font.h"
+//#include "../sid_tune.h"
+#include "UNIX.h"
+#include "../c64maze.h"
+#include <time.h>
+#include <stdbool.h>
+/* local definitions */
+#define VIC_II_START 53248U
```

Comment:
You can probably remove the constants related to the C64 that will not be useful for the bgi library and the z88dk.

Reply:
I got it from your Unix port (which surely doesn't need it as well) ,  I did it as a very quick hack but surely I can clean it up a little more (and provide the header file which is identical to the Unix one). 
To be precise this port is mostly for Turbo C, I've built a set of macros for z88dk to emulate emulating bgi.h, which partially relies on a  emulation layer for the X11 library (the proportional font).   The native graphics primitives for z88dk are slightly different and using them would enormously reduce the memory waste, but I though it was also nice to leave hints to easily get a driver for the Turbo C.